src/tokenizer.rs

```diff
@@ -300,6 +300,41 @@ impl fmt::Display for Whitespace {
     }
 }
 
+/// Location in input string
+#[derive(Debug, PartialEq, Clone)]
+pub struct Location {
+    /// Line number, starting from 1
+    pub line: u64,
+    /// Line column, starting from 1
+    pub column: u64,
+}
+
+/// A [Token] with [Location] attached to it
```

Comment:
We can probably avoid many of the changes in this PR by implementing `PartialEq` between Token and TokenWithLocation:

```rust
impl PartialEq<Token>` for TokenWithLocation
```

Which would allow things like

```rust
        if self.peek_nth_token(1).token == Token::RArrow {
```

to keep working